### PR TITLE
migrate pallet-session-benchmarking to bench V2 syntax

### PR DESCRIPTION
**Pull request by clangenb**
Migrates pallet-session-benchmarking to bench V2 syntax.

Part of:
* #6202

### CODE REVIEW

File: substrate/frame/session/benchmarking/src/mock.rs [hunk 1]
```diff
@@ -42,6 +42,7 @@ frame_support::construct_runtime!(
 		Balances: pallet_balances,
 		Staking: pallet_staking,
 		Session: pallet_session,
+		Historical: pallet_session::historical
 	}
```
Comment (clangenb) [inline substrate/frame/session/benchmarking/src/mock.rs:46]: It seems that benchmark v1 did not notice that the code doesn't compile for the extra.

File: substrate/frame/session/benchmarking/src/mock.rs [hunk 2]
```diff
@@ -79,7 +80,8 @@ sp_runtime::impl_opaque_keys! {
 
 pub struct TestSessionHandler;
 impl pallet_session::SessionHandler<AccountId> for TestSessionHandler {
-	const KEY_TYPE_IDS: &'static [sp_runtime::KeyTypeId] = &[];
+	// corresponds to the opaque key id above
+	const KEY_TYPE_IDS: &'static [KeyTypeId] = &[KeyTypeId([100u8, 117u8, 109u8, 121u8])];
```
Comment (clangenb) [inline substrate/frame/session/benchmarking/src/mock.rs:84]: This panicked in for `cargo test -p pallet-session-benchmarking --features runtime-benchmarks`, but it also failed on master. Is this not CI tested?

```console

---- mock::test_genesis_config_builds stdout ----
thread 'mock::test_genesis_config_builds' panicked at /home/clang/code/polkadot-sdk/substrate/frame/session/src/lib.rs:440:17:
Number of keys in session handler and session keys does not match
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

File: substrate/frame/session/benchmarking/src/mock.rs (hunk 1 above)
Comment (clangenb) [inline substrate/frame/session/benchmarking/src/mock.rs:46]: This failed on master:

```console
---- mock::test_genesis_config_builds stdout ----
thread 'mock::test_genesis_config_builds' panicked at /home/clang/code/polkadot-sdk/substrate/frame/session/src/historical/mod.rs:93:15:
No name_hash found for the pallet in the runtime! This usually means that the pallet wasn't added to `construct_runtime!`.
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

**Comment by clangenb** [comment]
@re-gius review bump :)

**Comment by re-gius** [comment]
bot bench substrate-pallet --features=runtime-benchmarks --pallet=pallet_session_benchmarking

**Review by re-gius — COMMENTED** [review]
Run `cargo +nightly fmt --all -- --check` (see CI https://github.com/paritytech/polkadot-sdk/actions/runs/11596200437/job/32286844067?pr=6294 )

**Comment by re-gius** [comment]
bot bench substrate-pallet --features=runtime-benchmarks --pallet=pallet_session

File: substrate/frame/session/benchmarking/src/mock.rs (hunk 2 above)
Comment (gui1117) [reply substrate/frame/session/benchmarking/src/mock.rs:84]: I opened an issue: https://github.com/paritytech/polkadot-sdk/issues/6726

**Review by gui1117 — APPROVED** [review]

**Review by re-gius — APPROVED** [review]

**Pull request merged**